doc/api/quic.md

```diff
@@ -0,0 +1,539 @@
+    will also display the name and description of each available elliptic curve.
+    **Default:** [`tls.DEFAULT_ECDH_CURVE`].
+  * `honorCipherOrder` {boolean} Attempt to use the server's cipher suite
+    preferences instead of the client's. When `true`, causes
+    `SSL_OP_CIPHER_SERVER_PREFERENCE` to be set in `secureOptions`, see
+    [OpenSSL Options][] for more information.
+  * `key` {string|string[]|Buffer|Buffer[]|Object[]} Private keys in PEM format.
+    PEM allows the option of private keys being encrypted. Encrypted keys will
+    be decrypted with `options.passphrase`. Multiple keys using different
+    algorithms can be provided either as an array of unencrypted key strings or
+    buffers, or an array of objects in the form `{pem: <string|buffer>[,
+    passphrase: <string>]}`. The object form can only occur in an array.
+    `object.passphrase` is optional. Encrypted keys will be decrypted with
+    `object.passphrase` if provided, or `options.passphrase` if it is not.
+  * `passphrase` {string} Shared passphrase used for a single private key and/or
+    a PFX.
+  * `pfx` {string|string[]|Buffer|Buffer[]|Object[]} PFX or PKCS12 encoded
+    private key and certificate chain. `pfx` is an alternative to providing
+    `key` and `cert` individually. PFX is usually encrypted, if it is,
+    `passphrase` will be used to decrypt it. Multiple PFX can be provided either
+    as an array of unencrypted PFX buffers, or an array of objects in the form
+    `{buf: <string|buffer>[, passphrase: <string>]}`. The object form can only
+    occur in an array. `object.passphrase` is optional. Encrypted PFX will be
+    decrypted with `object.passphrase` if provided, or `options.passphrase` if
+    it is not.
+  * `secureOptions` {number} Optionally affect the OpenSSL protocol behavior,
+    which is not usually necessary. This should be used carefully if at all!
+    Value is a numeric bitmask of the `SSL_OP_*` options from
+    [OpenSSL Options][].
+  * `sessionIdContext` {string} Opaque identifier used by servers to ensure
+    session state is not shared between applications. Unused by clients.
+
+Create a new `QuicClientSession`.
+
+### quicsocket.destroy([error])
+<!--YAML
+added: REPLACEME
+-->
+
+* `error` {any}
+
+Destroys the `QuicSocket` then emits the `'close'` event when done. The `'error'`
+event will be emitted after `'close'` if the `error` is not `undefined`.
+
+### quicsocket.destroyed
+<!--YAML
+added: REPLACEME
+-->
+
+* Type: {boolean}
+
+Will be `true` if the `QuicSocket` has been destroyed.
+
+### quicsocket.dropMembership(address, iface)
+<!--YAML
+added: REPLACEME
+-->
+
+* `address` {string}
+* `iface` {string}
+
+### quicsocket.fd
+<!--YAML
+added: REPLACEME
+-->
+
+* Type: {integer}
+
+The system file descriptor the `QuicSocket` is bound to.
+
+### quicsocket.listen([options][, callback])
+<!--YAML
+added: REPLACEME
+-->
+
+* `options` {Object}
```

Comment:
i think these options should be passed in when we create the socket.

Reply:
Yeah, still working out how all of the options are going to work. Should be tackling that in the next iteration.